lib/job.js

```diff
@@ -131,6 +131,28 @@ class Job extends BaseModel {
 
         return factory.list(listConfig);
     }
+
+    /**
+     * Remove all builds associated with this job and the job itself
+     * @return {Promise}        Resolves to null if remove successfully
+     */
+    remove() {
+        const removeBuilds = (() =>
+            // Iterate through the builds and remove them
+            this.getBuilds().then(builds => {
+                if (builds.length > 0) {
+                    builds.forEach(build => Promise.all(build.remove()));
```

Comment:
Wouldn't that just call getBuilds() once? I wanted to account for the scenario when there are like 200 builds, but getBuilds would just get the first 50 builds and remove them.